bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/DiscoveryResultBuilder.java

```diff
@@ -128,13 +129,24 @@ public DiscoveryResultBuilder withTTL(long ttl) {
     }
 
     /**
+     * Sets the class name of the {@link DiscoveryService} that has discovered this {@link DiscoveryResult}.
+     *
+     * @param service class name of the {@link DiscoveryService}
+     * @return the updated builder
+     */
+    public DiscoveryResultBuilder withDiscoveryService(String service) {
```

Comment:
I really don't like this idea. Have you checked how many bindings are using the DiscoveryResultBuilder directly...? Can't this be handles in the `AbstractDiscoveryService.thingDiscovered(...)` transparently, without the need to adapt bazillions of bindings?